src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.MemoryMappedInfo.cs

```diff
@@ -114,7 +121,10 @@ public Stream CreateReadableStream()
                         },
                         this);
                     streamAccessor = new ReferenceCountedDisposable<MemoryMappedViewAccessor>(rawAccessor);
-                    _weakReadAccessor = new ReferenceCountedDisposable<MemoryMappedViewAccessor>.WeakReference(streamAccessor);
+                    lock (this)
+                    {
+                        _weakReadAccessor = new ReferenceCountedDisposable<MemoryMappedViewAccessor>.WeakReference(streamAccessor);
+                    }
```

Comment:
woudl this be an issue for any other RCD usages in our codebase?

Reply:
I was not able to find any other cases suffering from the same problem. Following this pull request, the problem is not possible even without a lock.

Reply:
soudns good.  thanks for looking.